### ISSUE TITLE
3. Update to erlang 20.

### ISSUE THREAD

**Issue by PrajaktaPurohit** [chef/chef-server#1796]
Once the Chef Infra Server is upgraded to Erlang 19. Proceed with updating the dependencies to the latest versions and update erlang to version 20.

[x] Update the dependencies with rebar3 to the latest.

[x] Change software definition at https://github.com/chef/omnibus-software/blob/master/config/software/erlang.rb with version 20 of erlang.

[x] Update https://github.com/chef/chef-server/blob/master/omnibus_overrides.rb#L5 to the above updated version.

[x] Make the necessary code changes to oc_erchef, oc_bifrost, bookshelf.

[x] Update the buildkite pipeline to test with Erlang 20.

Acceptance Criteria:
[x] Verify pipeline passes with erlang 20.
[x] Adhoc version of Chef-server builds with Erlang 20.

**Comment by PrajaktaPurohit** [comment]
https://github.com/chef/chef-server/compare/lbaker/land-erlang20-bcrypt-rabbitmq2

**Comment by PrajaktaPurohit** [comment]
This branch is waiting on:
1. fixes to dialyzer
2. turn on warnings_as_errors
3. update hoax to latest
4. update rabbitmq to latest

**Comment by lbakerchef** [comment]
Commenting on #3 above "update hoax to latest..."

I forked the repo and fixed the dialyzer issue we had.  I would not recommend updating Hoax to the latest because in my humble opinion it really won't buy us anything:

1) it is an unmaintained repo (a dead end)
2) it is strictly used for mocking some stuff in tests (i.e. it isn't used on the running server)
3) it is only used in a few places, so it's not used a lot
4) where it is used, it is working
5) we are unlikely to expand our use of it and instead will probably rely on the other mocking library we are using
6) it will open up a can of worms with all the fixes we will need to make after updating it to the latest

Basically, if we update to the latest, we will incur pain for little to no gain, or even negative gain.  Having said that, this is simply me throwing in my 2 cents for consideration by the 'powers that be.'  I may not have considered everything.  And I'm certainly ready, willing, and able to update Hoax to the latest, if that's what we want to do.  As I used to say to a previous colleague, "the pay rate is the same no matter what we're doing!"  :)
